htlcswitch/switch_test.go

```diff
@@ -3754,9 +3754,22 @@ func assertOutgoingLinkReceiveIntercepted(t *testing.T,
 	}
 }
 
-func TestSwitchHoldForward(t *testing.T) {
-	t.Parallel()
+type interceptableSwitchTestContext struct {
```

Comment:
don't quite get the idea behind this test context, looks like in the end it will hold everything.

Reply:
The goal is to make the test context and methods operating within the context re-usable for the unit test that is added in the next commit.